# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/PrivateIterators-inl.h
# Commit: 054a08d13999
# Full Hash: 054a08d139993a42ff2de423e1dab2633baadba6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-04 04:18:28
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 5: Replace heap state checks that rely on there being a JSContext for the thread r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140192
# ==============================================================================

diff -r b29a261a06ec -r 054a08d13999 js/src/gc/PrivateIterators-inl.h
--- a/js/src/gc/PrivateIterators-inl.h	Thu Mar 03 17:49:17 2022 +0000
+++ b/js/src/gc/PrivateIterators-inl.h	Thu Mar 03 17:49:18 2022 +0000
@@ -51,7 +51,7 @@
 
  public:
   explicit GCZonesIter(GCRuntime* gc) : zone(gc) {
-    MOZ_ASSERT(JS::RuntimeHeapIsBusy());
+    MOZ_ASSERT(gc->heapState() != JS::HeapState::Idle);
     if (!done() && !zone->wasGCStarted()) {
       next();
     }